src/utils/model.js

```diff
@@ -1,157 +1,91 @@
-  try {
-    const data = await Promise.all(
-      urls.map(url =>
-        fetch(url)
-          .then(response => {
-            // 404 files currently return an html page instead of 404,
-            // so need to handle this way...
-            return response.text();
-          })
-          .then(responseText => {
-            try {
-              const jsonResponse = JSON.parse(responseText);
-
-              return jsonResponse;
-            } catch (err) {
-              throw err;
-            }
-          }),
-      ),
-    );
-
-    return data;
-  } catch (error) {
-    throw error;
-  }
+  return Promise.all(
+    urls.map(async url => {
+      try {
+        const response = await fetch(url);
+        const textResponse = await response.text();
+        return JSON.parse(textResponse);
+      } catch {
+        return null;
+      }
+    }),
+  );
 }
 
-export const ModelIds = {
-  baseline: 0,
-  strictDistancingNow: 1,
-  projected: 2,
-  weakDistancingNow: 3,
-};
-
-const initialData = {
-  location: null,
-  county: null,
-  stateDatas: null,
-  countyDatas: null,
-};
-
-async function fetchSummary(setModelDatas, location) {
-  const summaryUrl = `${DATA_URL}/county_summaries/${location.toUpperCase()}.summary.json`;
-
-  try {
-    const summary = await fetchAll([summaryUrl]);
-    setModelDatas(state => {
-      return {
-        ...state,
-        summary: summary[0],
-      };
-    });
-  } catch (err) {}
-}
-
-async function fetchData(location, county = null, dataUrl = DATA_URL) {
-  dataUrl = dataUrl || DATA_URL;
-  let modelDataForKey = null;
-  let projectionsToLoad;
-  if (!county) {
-    projectionsToLoad = [
-      ModelIds.baseline,
-      ModelIds.strictDistancingNow,
-      ModelIds.projected,
-      ModelIds.weakDistancingNow,
-    ];
-  } else {
-    projectionsToLoad = [
-      ModelIds.baseline,
-      ModelIds.strictDistancingNow,
-      ModelIds.strictDistancingNow, // TODO(igor): HAX!
-      ModelIds.weakDistancingNow,
-    ];
-  }
-  let urls = projectionsToLoad.map(i => {
-    let fipsCode =
-      county && county.full_fips_code ? county.full_fips_code : null;
-    const stateUrl = `${dataUrl}/${location}.${i}.json`;
-    const countyUrl = `${dataUrl}/county/${location.toUpperCase()}.${fipsCode}.${i}.json`;
-    return county ? countyUrl : stateUrl;
-  });
-
-  try {
-    let loadedModelDatas = await fetchAll(urls);
-    // HACK: Truncate data to 32 data points to make new models match old models.
-    loadedModelDatas = loadedModelDatas.map(data => data.slice(0, 32));
-    //This is to fix county data format
-    modelDataForKey = {
-      projections: new Projections(loadedModelDatas, location, county),
-    };
-  } catch (err) {
-    modelDataForKey = {
-      error: true,
-      payload: err,
-    };
-  }
-
-  return modelDataForKey;
-}
-
-async function fetchDataAndSet(
-  setModelDatas,
-  location,
-  county = null,
-  dataUrl = null,
+async function fetchProjections(
+  stateAbbr,
```

Comment:
nit: stateId?

Reply:
yea, let me touch it up in the second refactor